tensorflow-framework/src/test/java/org/tensorflow/framework/activations/ExponentialTest.java

```diff
@@ -60,7 +56,7 @@ public void testCallFloat() {
     for (TestSession.Mode tfMode : tfModes)
       try (TestSession session = TestSession.createTestSession(tfMode)) {
         Ops tf = session.getTF();
-        Exponential<TFloat32> instance = new Exponential<>(tf);
+        Exponential instance = new Exponential(tf);
```

Comment:
You're right. I will add a throws IllegalArgumentException.

```
org.tensorflow.exceptions.TFInvalidArgumentException: 
Value for attr 'T' of int32 is not in the list of allowed values: bfloat16, half, float, double, complex64, complex128
	; NodeDef: {{node Exp}}; Op<name=Exp; signature=x:T -> y:T; attr=T:type,allowed=[DT_BFLOAT16, DT_HALF, DT_FLOAT, DT_DOUBLE, DT_COMPLEX64, DT_COMPLEX128]>
	at org.tensorflow.framework.activations.ExponentialTest.testCallInt32(ExponentialTest.java:51)
```

Reply:
You should probably check the rest of them and add guards if necessary. If most of them throw, then this version is just moving type checking from compile time to runtime, and I'm not sure that that is a worthwhile tradeoff.

Reply:
I've been checking and many of the activations require `TFLoating`. I am beginning to think we should go back to the way  we had it.

BTW: There is a a new `Activation`, `GeLU`.

Reply:
Ah, well we definitely want `GeLU`, it's used in BERT.

Reply:
Yeah, I now believe that I led us astray by not fully understanding the restrictions on overriding generic methods versus overriding generic classes. I lean toward giving up on this path. Here's how I understand the situation:

With Jim's original approach, we'd subclass `Activation` while further restricting the generic parameter:
~~~ Java
public abstract class Activation<T extends TNumber> {
  . . .
  public abstract Operand<T> call(Operand<T> input);
}

public class ELU<T extends TFloating> extends Activation<T> {
  . . .
  @Override
  public Operand<T> call(Operand<T> input) {
    . . .
  }
}
~~~
`ELU<T>` is an `Activation<T>`, so it makes sense that this is a legal override. But also, we added the compile-time parameter restriction to `TFloating` that we need.

Our new approach doesn't give us the same flexibility, because the following doesn't compile:
~~~Java
public class Activation {
  . . .
    public abstract <T extends TNumber> Operand<T> call(Operand<T> input);
}

public class ELU extends Activation {
  . . .
  public <T extends TFloating> Operand<T> call(Operand<T> input) { // doesn't compile
  . . .
  }
}
~~~
With this approach, every `ELU` is an `Activation`, so `ELU`'s generic `call` method can't be more restrictive than `Activation`'s.

Reply:
Agreed. If we've shifted all the compile time checks into runtime checks then this is a bad idea for activations.

Reply:
The latest refactor of generic classes has addressed many of these issues.